app/controllers/user/lessons_controller.rb

```diff
@@ -1,7 +1,7 @@
 class User::LessonsController < User::ApplicationController
-  before_action :set_course, only: %i(show test_history)
-  before_action :set_lesson,
-                only: %i(show study test_history)
+  load_and_authorize_resource :course, class: "Course"
```

Comment:
```suggestion
  load_and_authorize_resource :course, class: Course.name
```
Tương tự cho các class khác